cmd/ci-operator/main.go

```diff
@@ -817,15 +826,74 @@ type prowResultMetadata struct {
 	JobVersion    string            `json:"job-version"`
 	Pod           string            `json:"pod"`
 	WorkNamespace string            `json:"work-namespace"`
+	Metadata      map[string]string `json:"metadata"`
 }
 
 func (o *options) writeMetadataJSON() error {
 	if len(o.artifactDir) == 0 {
 		return nil
 	}
 
-	m := prowResultMetadata{}
+	metadataJSONPath := filepath.Join(o.artifactDir, "metadata.json")
+
+	customProwMetadataFile := o.findCustomMetadataFile()
+
+	// If the metadata JSON exists and there's no custom prow metadata, then skip the second write.
+	_, err := os.Stat(metadataJSONPath)
+	if customProwMetadataFile == "" && err == nil {
+		log.Printf("No custom metadata found and prow metadata already exists. Not updating the metadata.")
+		return nil
+	}
+
+	var customMetadataErr error
+
+	m := o.generateProwMetadata()
+	m.Metadata, customMetadataErr = o.parseCustomMetadata(customProwMetadataFile)
+
+	if customMetadataErr != nil {
+		log.Printf("Error parsing custom metadata. (%v)", err)
+	}
+
+	data, err := json.MarshalIndent(m, "", "")
+
+	if o.dry {
+		log.Printf("metadata.json:\n%s", string(data))
+		return nil
+	}
+	err = ioutil.WriteFile(metadataJSONPath, data, 0640)
+
+	if err != nil {
+		return err
+	} else if customMetadataErr != nil {
+		return customMetadataErr
+	}
+
+	return nil
+}
+
+func (o *options) findCustomMetadataFile() (customProwMetadataFile string) {
+	// Try to find the custom prow metadata file. We assume that there's only one. If there's more than one,
+	// we'll just use the first one that we find.
+	filepath.Walk(o.artifactDir, func(path string, info os.FileInfo, err error) error {
```

Comment:
This returns an error you're ignoring -- I would suggest looking to see if your editor can highlight that